### DISCUSSION TITLE
Colab for Rave2?

### DISCUSSION THREAD

**Discussion by futorio** [acids-ircam/RAVE#162] [General]
Maybe add google Colab notebook for training rave v2?

**Comment by JulienVincenot** [comment]
Is there a way to modify the old Colab to still use v1? (temporarily? I'd still need to use the prior)

**Reply by futorio** [reply]
I'll try to port [this](https://colab.research.google.com/drive/1aK8K186QegnWVMAhfnFRofk_Jf7BBUxl) existing notebook

**Comment by JulienVincenot** [comment]
@caillonantoine Any suggestion how to modify the original colab to use v1? It's been broken for a few days...
I guess there's a way to refer to the older archive on github instead of just getting the latest commit?

And/or are you planning to share a colab for v2 version soon?

(I have no idea how difficult that is to make, so with all patience and understanding of course. I'm just hoping it comes out soon I was planning to use a lot of rave for a project coming quite soon and I have no hardware alternative at the moment 😅 I would still be happy with a functioning V1 on colab.

Thanks !

**Reply by caillonantoine** [reply]
Locate the `git clone ...` in the installation part of the colab and add a `-b v1`, it should do the trick 🙂

**Reply by JulienVincenot** [reply]
Ok it worked thanks Antoine. I assume "build" v1, easier than I thought. 
I should just rtfm but terminal Git is still a bit intimidating... 😅

**Reply by caillonantoine** [reply]
It is actually `branch` ! If you want to clone a repository using a specific branch you can pass `-b master` or `-b release` or a tag number :)

**Reply by JulienVincenot** [reply]
oh indeed thx ! 🤦‍♂️ holy sh** that git man page is a like a maze 😵

**Reply by JulienVincenot** [reply]
@caillonantoine sorry to bother you again with that old version, but I finished training a model on the old Colab file, adding -b v1 as you suggested, and as I tried to start training the prior like before now for this new run I get this error : 

The code :
```
%cd /content/drive/MyDrive/RAVE_COLLAB/
!/content/miniconda/bin/python train_prior.py --pretrained-vae rave_{training_name}.ts --preprocessed /content/preprocessed/prior --wav /content/drive/MyDrive/RAVE_COLLAB/resampled/$training_name/ --name $training_name --val-every $validation_every
```
Error message :
```
/content/drive/MyDrive/RAVE_COLLAB
  File "/content/drive/MyDrive/RAVE_COLLAB/train_prior.py", line 81
    num_workers = 0 is os.name == "nt" else 8
                                       ^
SyntaxError: invalid syntax
```
any idea what I could be missing? is it possible that the prior was disabled also in that branch? 
Thanks !

**Reply by JulienVincenot** [reply]
hi @caillonantoine not sure if you saw my last post, I'm still stuck with the prior on V1 for some reason. 
Here are a few screenshots perhaps to help identify the issue. See my message above for the last command (for the prior) and error message I get. Do you have any idea that could help? Thanks in advance !


<img width="1208" alt="Capture d’écran 2023-02-07 à 16 46 26" src="https://user-images.githubusercontent.com/18226950/218417413-59e0da9e-df72-4383-bfaa-d4d7017b6f42.png">
<img width="1208" alt="Capture d’écran 2023-02-07 à 16 47 05" src="https://user-images.githubusercontent.com/18226950/218417397-51f1efd4-6882-4719-9bd5-84e6ad3dc5ca.png">
<img width="1189" alt="Capture d’écran 2023-02-07 à 16 46 11" src="https://user-images.githubusercontent.com/18226950/218417354-f9226d5c-2b30-429c-9524-92c897b1aa74.png">
?

**Comment by moiseshorta** [comment]
Here's a no-frillz notebook for training RAVE v2 collab:

https://colab.research.google.com/drive/1ih-gv1iHEZNuGhHPvCHrleLNXvooQMvI?usp=sharing

Cheers!

**Reply by caillonantoine** [reply]
Thanks for that ! Do you mind if I link your colab to the README ?

**Reply by moiseshorta** [reply]
> Thanks for that ! Do you mind if I link your colab to the README ?

Sure thing, please go ahead!

**Reply by plagasul** [reply]
Thanks for the collab, ¡gracias! would there be any tutorial or in-depth article that could be used to learn our way with setting up a dataset for Rave that could be used with the collab? Thank you.
